lib/srv/app/session.go

```diff
@@ -113,13 +148,8 @@ func (s *Server) newStreamWriter(identity *tlsca.Identity, app types.Application
 		return nil, trace.Wrap(err)
 	}
 
-	// Each chunk has its own ID. Create a new UUID for this chunk which will be
-	// emitted in a new event to the audit log that can be use to aggregate all
-	// chunks for a particular session.
-	chunkID := uuid.New().String()
-
 	// Create a sync or async streamer depending on configuration of cluster.
-	streamer, err := s.newStreamer(s.closeContext, chunkID, recConfig)
+	streamer, err := s.newStreamer(s.closeContext, identity.RouteToApp.SessionID, recConfig)
```

Comment:
Why did the chunk ID change from the new UUID for session ID?

Reply:
I see, it looks like you're right, thanks for the pushback.

Do you know why it was designed this way? it seems like keeping the session recording together in a single `.tar` would be simpler to follow, playback, etc.

Reply:
(Quickly jumping here to request a comment - this is bound to surprise others too.)

Reply:
The problem is this won't work at all with the session tracker logic in the session upload completer.

Reply:
It looks like the intention of this design choice was to keep application services stateless - https://github.com/gravitational/teleport/blob/master/rfd/0008-application-access.md#audit-events

Reply:
@Joerger If I recall correctly, this was done because in app access there's no clear indication of when a "session" is completed. Because it's just HTTP. So we have to know when to upload.

Also keep in mind that in HA mode the request can land on any app agent which serves the application so we wouldn't be able to assign the same ID to chunks anyway.

Reply:
It looks like the intention of this design choice was to keep application services stateless - https://github.com/gravitational/teleport/blob/master/rfd/0008-application-access.md#audit-events.

If I'm understanding it correctly, the goal of using chunks was to consistently send 5 minute upload chunks to the auth server instead of waiting until the end of the session to send the full session recording, to prevent a backlog of long standing session uploads from sitting on the application service.

If that's the case, this is essentially just `node-sync`, but with a more confusing user/developer experience and without the benefits of actually being synchronous, aka actually being stateless rather than having a short lived state.

@r0mant I'm not sure what to do, here are a couple paths I see:
1) Remove the session chunk logic from application access. Instead the session will be uploaded the same way other uploads are - `logs/records/<session_id>.tar`.
  - If we need to maintain the "statelessness" of proxy and app services with application recordings, we could always use `sync` recording regardless of what's set in auth config.
  - There are obviously some backwards compatibility concerns here. What I would propose is we upload both the old session chunks and the new single session recording. With https://github.com/gravitational/teleport/pull/12471, The old session chunk strategy will continue to work with the 24 hour grace period, while the new single session recording will work with the session tracker logic.
    - While we're at it we could add `tsh app play` since it isn't currently documented, at least in a user friendly way, how to play app session recordings.
 2) Add session trackers for every app session chunk. With this choice, we should commit to treating `chunk_id` as the actual "session" from the developer point of view. We can add `AppSessionID` to session trackers to point to the "SessionID" which is help in application access certificates.

Reply:
> @Joerger If I recall correctly, this was done because in app access there's no clear indication of when a "session" is completed. Because it's just HTTP. So we have to know when to upload.
> 
> Also keep in mind that in HA mode the request can land on any app agent which serves the application so we wouldn't be able to assign the same ID to chunks anyway.

`identity.RouteToApp.SessionID` points to the sessionID held in the application access cert, so even in HA mode the same sessionID could be used.

Also the HA use case points further to require `sync` recording, since the AuthServer would be able to serve multiple application access agents with the same sessionID and string the uploads together. 

If I'm understanding correctly, the current HA recording could lead to:
```
Chunk 1 (app agent 4):
   - event 1
   - event 3
   
Chunk 2 (app agent 2):
  - event 2
  - event 4
```

With sync, we should always get:
```
Session Upload:
   event 1
   event 2
   event 3
   event 4
```

Reply:
@Joerger Option "2" (having a session tracker per chunk) seems less invasive and less prone to regressions IMO.